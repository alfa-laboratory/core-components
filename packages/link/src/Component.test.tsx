import React from 'react';
import { render } from '@testing-library/react';

import { Link } from './index';

describe('Snapshots tests', () => {
    it('should match snapshot', () => {
        expect(render(<Link href=''>Link</Link>)).toMatchSnapshot();
    });
});

describe('Classes tests', () => {
    it('should set custom class', () => {
        const className = 'custom-class';

        const { container } = render(
            <Link className={className} href=''>
                Link
            </Link>,
        );

        expect(container.firstElementChild).toHaveClass(className);
    });

    it('should set view `primary` as default view', () => {
        const { container } = render(<Link href=''>Link</Link>);

        expect(container.firstElementChild).toHaveClass('primary');
    });

    it('should set view', () => {
        const { container } = render(
            <Link href='' view='default'>
                Link
            </Link>,
        );

        expect(container.firstElementChild).toHaveClass('default');
    });

    it('should set `pseudo` class if prop `pseudo` is present', () => {
        const { container } = render(
            <Link href='' pseudo={true}>
                Link
            </Link>,
        );

        expect(container.firstElementChild).toHaveClass('pseudo');
    });
});

describe('Attributes tests', () => {
    it('should set data-test-id attribute', () => {
        const dataTestId = 'link-test-id';

        const { container } = render(
            <Link href='' dataTestId={dataTestId}>
                Link
            </Link>,
        );

        const testIdAttr = container.firstElementChild?.getAttribute('data-test-id');

        expect(testIdAttr).toBe(dataTestId);
    });
});

describe('Render tests', () => {
    it('should unmount without errors', () => {
        const { unmount } = render(<Link href=''>Link</Link>);

        expect(unmount).not.toThrowError();
    });
});
