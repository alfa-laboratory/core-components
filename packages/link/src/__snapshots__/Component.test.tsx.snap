// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots tests should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="component primary"
        href=""
      >
        <span>
          <span
            class="text"
          >
            Link
          </span>
        </span>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="component primary"
      href=""
    >
      <span>
        <span
          class="text"
        >
          Link
        </span>
      </span>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshots tests should render left slot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="component primary"
        href=""
      >
        <span
          class="addons"
        >
          <svg
            fill="currentColor"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              d="M9.259 8.131L12 1v16l-7 4 2.32-7.12.094-.275-.243-.18L1 8.5l7.88-.097h.287l.092-.272zm5.482 0L12 1v16l7 4-2.32-7.12-.094-.275.243-.18L23 8.5l-7.88-.097h-.287l-.092-.272z"
            />
          </svg>
        </span>
        <span>
          <span
            class="text"
          >
            Link
          </span>
        </span>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="component primary"
      href=""
    >
      <span
        class="addons"
      >
        <svg
          fill="currentColor"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M9.259 8.131L12 1v16l-7 4 2.32-7.12.094-.275-.243-.18L1 8.5l7.88-.097h.287l.092-.272zm5.482 0L12 1v16l7 4-2.32-7.12-.094-.275.243-.18L23 8.5l-7.88-.097h-.287l-.092-.272z"
          />
        </svg>
      </span>
      <span>
        <span
          class="text"
        >
          Link
        </span>
      </span>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshots tests should render right slot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="component primary"
        href=""
      >
        <span>
          <span
            class="text"
          >
            Link
          </span>
        </span>
        <span
          class="addons"
        >
          <svg
            fill="currentColor"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              d="M9.259 8.131L12 1v16l-7 4 2.32-7.12.094-.275-.243-.18L1 8.5l7.88-.097h.287l.092-.272zm5.482 0L12 1v16l7 4-2.32-7.12-.094-.275.243-.18L23 8.5l-7.88-.097h-.287l-.092-.272z"
            />
          </svg>
        </span>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="component primary"
      href=""
    >
      <span>
        <span
          class="text"
        >
          Link
        </span>
      </span>
      <span
        class="addons"
      >
        <svg
          fill="currentColor"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M9.259 8.131L12 1v16l-7 4 2.32-7.12.094-.275-.243-.18L1 8.5l7.88-.097h.287l.092-.272zm5.482 0L12 1v16l7 4-2.32-7.12-.094-.275.243-.18L23 8.5l-7.88-.097h-.287l-.092-.272z"
          />
        </svg>
      </span>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
